lib/parser/demunge.ts

```diff
@@ -11,10 +12,19 @@ export default demunge;
 /**
  * Demunges the given policy object.
  * @param policy  The policy object to demunge
- * @param apiRoot The root URL of the Snyk API
+ * @param baseUrlConfig The URL config for the Snyk API, if the apiRoot will not be the same for all vulnerabilities,
+ * eg. some of them are licenses, in that case customUrlMap can be provided for vulnerability id regex matching
+ *  * Example:
+ *
+ *  const baseUrlConfig: BaseUrlConfig = {
+ *    apiRoot: '[https://security.snyk.io](https://security.snyk.io)',
+ *    customUrlMap: new Map([
+ *      [new RegExp(/^snyk:lic/i), '[https://snyk.io](https://snyk.io)'],
+ *    ]),
+ *  };
  * @returns The demunged policy object
  */
-function demunge(policy: Policy, apiRoot = '') {
+function demunge(policy: Policy, baseUrlConfig?: BaseUrlConfig) {
```

Comment:
Could we avoid the breaking change by still accepting a static API root? And also, wondering if a function would be more  flexible than a regex map. So something like: `apiRoot: string | (string) -> string`

Reply:
I was considering that approach but the only consumer that is using that function with the `apiRoot` parameter actually needs to be updated to pass the map as well 🤷‍♀️

Reply:
`demunge` is a public function though, so we don't know who it would be breaking for

Reply:
Ah outside of snyk, that is good point, ok

Reply:
Done!